pandas/core/internals/managers.py

```diff
@@ -1780,23 +1780,25 @@ def form_blocks(arrays, names, axes):
 
     if len(items_dict["CategoricalBlock"]) > 0:
         cat_blocks = [
-            make_block(array, klass=CategoricalBlock, placement=[i])
+            make_block(array, klass=CategoricalBlock, placement=slice(i, i + 1, 1))
             for i, _, array in items_dict["CategoricalBlock"]
         ]
         blocks.extend(cat_blocks)
 
     if len(items_dict["ExtensionBlock"]):
 
         external_blocks = [
-            make_block(array, klass=ExtensionBlock, placement=[i])
+            make_block(array, klass=ExtensionBlock, placement=slice(i, i + 1, 1))
             for i, _, array in items_dict["ExtensionBlock"]
         ]
 
         blocks.extend(external_blocks)
 
     if len(items_dict["ObjectValuesExtensionBlock"]):
         external_blocks = [
-            make_block(array, klass=ObjectValuesExtensionBlock, placement=[i])
+            make_block(
+                array, klass=ObjectValuesExtensionBlock, placement=slice(i, i + 1, 1)
+            )
```

Comment:
rather than changing this here
simply convert a single integer into a slice at a lower level

Reply:
Ah, indeed, I can create the slice inside BlockPlacement constructor. But don't we then want to explicitly pass a single integer instead of a list of 1 integer?

Reply:
we require a list / slice for placement

Reply:
we require list / slice

Reply:
To compare, I just pushed a commit that does both. 
Personally, I like passing it as an integer. It's another 33% faster as passing it as a slice or 1-element list, and it makes it also explicit when constructing it that it is about a single column.

Reply:
Whoops, I missed you proposed a single integer yourself (for some reason I thought you wanted to catch the single element list in BlockPlacement). Updated.